### PR TITLE
Model tracking

### LINKED ISSUE #617: Model Tracking
It is a common practice for data scientists to train tens of temporary models before picking one of them as their candidate model for going into production.
A model tracking tool can help each individual data scientist to easily track the models trained for an experiment (project) and compare their metrics for choosing the best one.

### PR DESCRIPTION
**Pull request by afard**
We introduce a new Python class, named vExperiment, to track the desired models.

This PR closes #617

### CODE REVIEW

File: verticapy/mlops/model_tracking/base.py [hunk 1]
```diff
@@ -0,0 +1,636 @@
...
+            for model_index, item in enumerate(self.user_defined_metrics_):
+                if (metric in item.keys()) and (parameter in self.parameters_[model_index].keys()):
+                    parameter_list.append(self.parameters_[model_index][parameter])
+                    metric_list.append(item[metric])
+
+        if len(parameter_list) == 0 or len(metric_list) == 0:
+            raise ValueError("Could not find any datapoint for the provided pair of " \
+                             "(parameter, metric) = ({parameter}, {metric}).")
+
+        data = {"X": np.concatenate(parameter_list, metric_list), "s": None, "c": None}
```
Comment (afard) [inline verticapy/mlops/model_tracking/base.py:447]: This doesn't work properly and I need help on this.

Comment (afard) [reply verticapy/mlops/model_tracking/base.py]: Thanks Umar! It works now.

**Comment by afard** [comment]
> [CLA assistant check](https://cla-assistant.io/vertica/VerticaPy?pullRequest=670) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/vertica/VerticaPy?pullRequest=670) before we can accept your contribution.**1** out of **2** committers have signed the CLA.✅ afard❌ Arash Fard
> 
> **Arash Fard** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vertica/VerticaPy?pullRequest=670) it.

I pushed a commit from my Windows in which the email for Git was not set. That is why the contributor is not recognized as me. Unfortunately, there is no easy way for linking an old commit to my account. So let's ignore the CLA requirement for this commit.

File: verticapy/tests_new/mlops/test_model_tracking.py [hunk 2]
```diff
@@ -0,0 +1,664 @@
...
+
+import verticapy.mlops.model_tracking as mt
+import verticapy.sql.sys as sys
+
+set_option("print_info", False)
+
+
+class ExperimentBase:
+    """
+    A base clase for model tracking tests with different experiment types
```
Comment (mail4umar) [inline verticapy/tests_new/mlops/test_model_tracking.py:36]: type: class

**Review by mail4umar — APPROVED** [review]

File: verticapy/tests_new/mlops/test_model_tracking.py (hunk 2 above)
Comment (afard) [reply verticapy/tests_new/mlops/test_model_tracking.py]: Fixed!

**Comment by afard** [comment]
> > [CLA assistant check](https://cla-assistant.io/vertica/VerticaPy?pullRequest=670) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/vertica/VerticaPy?pullRequest=670) before we can accept your contribution.**1** out of **2** committers have signed the CLA.✅ afard❌ Arash Fard
> > **Arash Fard** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> > You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vertica/VerticaPy?pullRequest=670) it.
> 
> I pushed a commit from my Windows in which the email for Git was not set. That is why the contributor is not recognized as me. Unfortunately, there is no easy way for linking an old commit to my account. So let's ignore the CLA requirement for this commit.

I added my microfocus email to my account and it resolved the CLA requirement.

**Pull request merged**